Automattic/jetpack
# Tests: Fix block performance tests

This PR switches block performance tests to use `lodash-es` instead of `lodash`.

## Proposed changes:


The [last successful run](https://github.com/Automattic/jetpack/actions/runs/5897193779) of block performance tests was on 17 August 2023 (over a year ago). It continued to run every 12 hours, but obviously hasn't been monitored.

The error, at least most recently, is as follows:

```
SyntaxError: Named export 'isArray' not found. The requested module 'lodash' is a CommonJS module, which may not support all module.exports as named exports.
```

An example of a recent failure:
https://github.com/Automattic/jetpack/actions/runs/11365267219

I'm guessing the breakage was due to a node version update no longer allowing CommonJS imports into ES modules or a repackaging of `lodash`, but I wasn't able to find any such thing.

I attempted a few solutions:

* `const { isArray, mergeWith } = require('lodash');`
* `import _ from 'lodash'`
* `import mergeWith from 'lodash/mergeWith';`

The former didn't work because the script is strictly ESM, and the latter two gave this error:

```
ESLint: Import members from the full Lodash module.(lodash/import-scope)
```

In the end, I just swapped out the library for the ESM-friendly variant.

Out of scope: test/script improvements. I just focused on getting it running again.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






You can manually trigger the hour-long workflow here (make sure to run against this branch):

https://github.com/Automattic/jetpack/actions/workflows/block-performance.yml

A recent successful run can be seen here:

https://github.com/Automattic/jetpack/actions/runs/11370883494

And notifications can be seen here:

p1729098723198299-slack-C1Y1F7ZM1
